File: 3605-construct-the-minimum-bitwise-array-i/3605-construct-the-minimum-bitwise-array-i.cpp
class Solution {
public:
       vector<int> minBitwiseArray(vector<int>& nums) {
        vector<int> ans;
        for (int n : nums) {
            int bit = 0;
            while (n & (1 << bit))                              // Find the first 0 bit
                bit++;
            ans.push_back(n > 2 ? n ^ (1 << (bit - 1)) : -1);   // Flip the previous 1 bit
        }
        return ans;
    }
};

File: 3605-construct-the-minimum-bitwise-array-i/3605-construct-the-minimum-bitwise-array-i.java
class Solution {
    public int[] minBitwiseArray(List<Integer> nums) {
        int[] ans = new int[nums.size()];
        int ind = 0;
        for(Integer val : nums)
        {
            int bit = 0;
            while((val & (1 << bit)) !=  0)
            {bit++;}
            ans[ind] = val > 2 ? (val ^ (1 << (bit - 1)) ) : -1;
            ind++;
        }

        return ans;
    }
}